truffle/src/main/java/org/jruby/truffle/om/dsl/api/Layout.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.om.dsl.api;
+
+import com.oracle.truffle.api.object.DynamicObject;
+import org.jruby.truffle.om.dsl.processor.OMProcessor;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A {@link Layout} annotation is attached to an interface that describes an
+ * object layout with properties. The interface has a factory method, guards,
+ * getters and setters. From this a class is generated that implements these
+ * methods and provides very efficient static access to these properties in a
+ * {@link DynamicObject}.
+ *
+ * <pre>
+ * @@Layout
+ * public interface RectLayout {
+ *    ...
+ * }
+ * </pre>
+ *
+ * The properties are defined by getters and setter method pairs. They should
```

Comment:
Minor issue, but I'd think "getters" should be plural here if used as a component of the pair description.